hack/setup-envtest.sh

```diff
@@ -0,0 +1,66 @@
+#!/usr/bin/env bash
+
+#  Copyright 2020 The Kubernetes Authors.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+set -e
+
+function setup_envtest_env {
+  # Setup env vars
+  if [[ -z "${KUBEBUILDER_ASSETS}" ]]; then
+    export KUBEBUILDER_ASSETS=$1/bin
+  fi
```

Comment:
```suggestion
  if [[ -z "${KUBEBUILDER_ASSETS}" ]]; then
    header_text "setting up kubebuilder-tools@${k8s_version} env vars"
    export KUBEBUILDER_ASSETS=$1/bin
  fi
```

Reply:
Is KUBEBUILDER_ASSETS preferable to what we were doing before

```
export PATH=${tmp_root}/kubebuilder/bin:$PATH
  export TEST_ASSET_KUBECTL=${tmp_root}/kubebuilder/bin/kubectl
  export TEST_ASSET_KUBE_APISERVER=${tmp_root}/kubebuilder/bin/kube-apiserver
  export TEST_ASSET_ETCD=${tmp_root}/kubebuilder/bin/etcd
```

Reply:
I removed calls to `header_text` because that lives in `common.sh`, and I didn't want to force users to download two scripts for this.

If the header text is important, I could copy that code into this script. But I figured callers of these functions could call `header_text`.

I opted to keep the [existing environment setup as what's in `master`](https://github.com/kubernetes-sigs/controller-runtime/blob/b4e68515dac4a6469f1992031e2e2d07b362b456/hack/common.sh#L55-L62), but I'm not opposed to making this change.

Reply:
Yeah let's copy it, it's great to see some styled output when running it